pkg/authconfig/authconfig_handler_test.go

```diff
@@ -0,0 +1,124 @@
+package authconfig_test
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"testing"
+
+	"github.com/codeready-toolchain/registration-service/pkg/authconfig"
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/gin-gonic/gin"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestAuthClientConfigHandler(t *testing.T) {
+	// Create a request to pass to our handler. We don't have any query parameters for now, so we'll
+	// pass 'nil' as the third parameter.
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/authconfig", nil)
+	require.NoError(t, err)
+
+	// Create logger and registry.
+	logger := log.New(os.Stderr, "", 0)
+	configRegistry := configuration.CreateEmptyRegistry()
+
+	// Set the config for testing mode, the handler may use this.
+	configRegistry.GetViperInstance().Set("testingmode", true)
```

Comment:
I think it would be useful if we set the testingmode globally for our tests instead of setting it explicitly in every test which requires that mode.

There are multiple ways to achieve that. We could create a test suite (see https://godoc.org/github.com/stretchr/testify/suite) which we can re-use in all relevant tests and which will do some initialization for every test in the suite.

For example in `test` package:
```
type UnitTestSuite struct {
    suite.Suite
    Config *configuration.Registry
}

func (s *UnitTestSuite) SetupSuite() {
    s.Config = ... // create a config and set it up
}

func (s *UnitTestSuite) TearDownSuite() {
	s.Config = nil // Summon the GC!
}
```

Then we can use suite our tests:

```
type TestAuthClientConfigSuite struct {
	test.UnitTestSuite
}

func TestRunAuthClientConfigSuite(t *testing.T) {
	suite.Run(t, &TestAuthClientConfigSuite{&test.UnitTestSuite{}})
}

func (s *TestAuthClientConfigSuite) TestHandler() {
	// You can test the handler here. The configuration is available in s.Config
}
```

Reply:
Makes sense, let's do this in a seperate PR. I created one with the base TestSuite and the AuthClientTest as an example. We can update all existing tests in that PR: https://github.com/codeready-toolchain/registration-service/pull/24